pandas/io/json/_json.py

```diff
@@ -814,9 +827,45 @@ def __init__(
             self.nrows = validate_integer("nrows", self.nrows, 0)
             if not self.lines:
                 raise ValueError("nrows can only be passed if lines=True")
+        if self.engine == "pyarrow":
+            if not self.lines:
+                raise ValueError(
+                    "currently pyarrow engine only supports "
+                    "the line-delimited JSON format"
+                )
 
-        data = self._get_data_from_filepath(filepath_or_buffer)
-        self.data = self._preprocess_data(data)
+        if self.engine == "pyarrow":
+            self._engine = self._make_engine(filepath_or_buffer, self.engine)
+        if self.engine == "ujson":
+            data = self._get_data_from_filepath(filepath_or_buffer)
+            self.data = self._preprocess_data(data)
+
+    def _make_engine(
+        self,
+        filepath_or_buffer: FilePath | ReadBuffer[str] | ReadBuffer[bytes],
+        engine: JSONEngine = "pyarrow",
+    ) -> ArrowJsonParserWrapper:
+
+        if not isinstance(filepath_or_buffer, list):
+            is_text = False
+            mode = "rb"
+            self.handles = get_handle(
+                filepath_or_buffer,
+                mode=mode,
+                encoding=self.encoding,
+                is_text=is_text,
+                compression=self.compression,
+                storage_options=self.storage_options,
+                errors=self.encoding_errors,
+            )
+            filepath_or_buffer = self.handles.handle
+
+        try:
+            return ArrowJsonParserWrapper(filepath_or_buffer)
+        except Exception:
```

Comment:
Generally should avoid `except Exception` and catch the specific one that should raise

Reply:
@mroeschke I took this piece of code from the `_make_engine` method in `read_csv`, therefore I don't know which exception it is trying to catch.

Reply:
I think then this should be a `finally` block and let exceptions raise as normal

Reply:
This still needs to be addressed